ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -222,4 +225,50 @@ private BackupReplayFile newReplayFile() {
         ? Optional.of(new BackupReplayFile(latestBakFile))
         : Optional.empty();
   }
+
+  private void ensureDirectoryExists(final File backupsDir) {
```

Comment:
nit: can we de-dup this code from `enforceStreamStateDirAvailability()` in `KsqlServerMain`? Not required if it's annoying to de-dup (especially since we're trying to keep this patch small in order to cherry-pick) but it's weird to me that we have the same code in two places.

Reply:
Right, I see it. But I have the extra step to restrict the permissions to the KSQL server user only. I'd like to do that in the state directory too, but I don't know why the directory is not too restrictive. I'd rather investigate more and do it in another follow-up PR.